### DISCUSSION TITLE
Audiomoth microphone not working?

### DISCUSSION THREAD

**Discussion by diplopoda** [rdz-oss/BattyBirdNET-Pi#15] [General]
Hey,

at the moment I'm not sure if I'm too dumb to understand the settings or the Audiomoth is not working correctly.

The Audiomoth is on the USB firmware 1.2.4 and I have a fresh installation of BattyBirdNET-Pi running on an RPI 4 B.

I'm not sure if the BattyBirdNET is not recognising the Audiomoth and it seems I can't find any setting or status in the tools to check what the problem is.

Big thanks in advance. BR, Robert

**Comment by rdz-oss** [comment]
Hi Robert,

for the audiomoth there is hope - working for many. 
Do you have the usb version or the one with the batteries?
One can
- Use the audiomoth configuration software on a pc to set the system to 250kHz . You seem to already have flashed it (settings also?)
- Set the little hardware lever to 'custom'
Have you tried that (setting custom on harware lever is sometimes overlooked)
Let me know if it works or we can work though some other hypothesis.

BR
Richard

**Reply by rdz-oss** [reply]
You can check the audio cards on the command of the Pi by
cat /proc/asound/cards

Only if above not works ('custom' hardware lever setting at least on usb audiomoth)
Find the audio card with above command ( a number)
set it (replace 1 by the number  you identified before)
sudo nano /etc/asound.conf
and edit there:
defaults.pcm.card 1
defaults.ctl.card 1

**Reply by diplopoda** [reply]
Hello Richard,

I have the "normal" version with batteries. Never used it with the USB firmware before and my problem was that I assumed it has to be in USB mode on the hardware lever.

The Audiomoth is recognised, I see the spectrogram and hear sound. But it seems the BattyBirdNet-Pi is not recording anything. Despite the fact that there are Ppip flying.

Also some menus of the webinterface don't show up (View Log, Web Terminal) and some processes are not active:

arecord	OFF
avahi-daemon	ON
ffmpeg	ON
gotty	OFF
icecast2	ON
python3	OFF
sshd	ON


BirdNET_6K_GLOBAL_MODEL (2020) is the correct model for the BattyBirdNet-Pi?

At the moment I'm just not sure if the BattyBirdNet-Pi is running correctly. No recordings at all - not even a false recording.

Thanks - Robert

**Reply by rdz-oss** [reply]
Hi Robert,

first of all good news the mic issue is resolved. For the other ones, I am not sure. Can you access the Tools menu? (use birdnet as username, no password). You can check the Services tab and see what is running and what not. Try using 'Stop core services', after that  'Restart services' and wait a minute. 

Let me know what happens. There is hope- Pip pip is the most common bat in the training set and is registered on my setup regularly. 

Let me know what the system does after above steps. It is odd that the Web terminal does not show up. Do you mean the Log View? Or the web terminal to log into the system via the browser?

**Reply by rdz-oss** [reply]
This is what Bavaria classifies as kuhls in my backyard from tonight :), perhaps you can have a look ?
<img width="935" alt="Screenshot 2024-03-17 at 20 23 42" src="https://github.com/rdz-oss/BattyBirdNET-Pi/assets/63958211/8e60decf-20f7-4b87-a62b-d02e0a4ce56d">

And social call (song type D ?)

<img width="878" alt="Screenshot 2024-03-17 at 20 24 57" src="https://github.com/rdz-oss/BattyBirdNET-Pi/assets/63958211/c6b1dc8d-dda4-4f8c-83de-bf0cac1104fd">

**Comment by diplopoda** [comment]
Yes, Tools is working. Stopped the core services and this appeared:

![batpi2](https://github.com/rdz-oss/BattyBirdNET-Pi/assets/2921983/99e7739a-6092-4be6-a452-b5d34e8a15aa)

After opening Services again:

![batpi3](https://github.com/rdz-oss/BattyBirdNET-Pi/assets/2921983/f725554d-0d36-409b-98bb-dcf2a245d285)

I tried "Restart services" again and the BattyBirdNet-Pi was not responding, restarted via SSH.

After restart all Services are "active" in the Services tab.

The main Overview looks like this and I'm not sure if the "missing image" on the stats left is normal.

![batpi4](https://github.com/rdz-oss/BattyBirdNET-Pi/assets/2921983/c61e9574-fdd0-4c18-b216-23bb641d72e8)


When I click on "View Log" it looks like this:

![batpi5](https://github.com/rdz-oss/BattyBirdNET-Pi/assets/2921983/14688b8c-87b6-4466-9209-a16bd69dd2a4)

The same when I click on "Web Terminal" in Tools:

![batpi6](https://github.com/rdz-oss/BattyBirdNET-Pi/assets/2921983/b981413d-8b8e-49c3-8292-d175c5c3f0a0)

Thanks, Robert

**Reply by rdz-oss** [reply]
Hi Robert. To me it looks like something is broken on the system. You should at least see the incoming spectrogram and a terminal output for the View Log. Likely something went wrong in the install. 
Consider reinstalling. 
Raspi 4B you said
on a legacy 64 bit raspian lite (? - latest version not supported as yet)
run a fresh install
check if it ends with success messages.

Sorry for your trouble, but its pretty cool once it is setup!

BR
Richard

**Comment by diplopoda** [comment]
Hello Richard,

hmm ok.

It is a Raspberry Pi 4 Model B with 8GB RAM.

I used this images:

Raspberry Pi OS (Legacy) Lite
Release date: March 12th 2024
System: 64-bit
Kernel version: 6.1
Debian version: 11 (bullseye)

Will do a fresh install after you confirmed that this should work with this combination ;)

Thanks - Robert

**Comment by diplopoda** [comment]
Ok, tried to make a fresh install on a new SD card etc.:

![batpi_error](https://github.com/rdz-oss/BattyBirdNET-Pi/assets/2921983/2abc43a6-8970-428a-8584-f77a5b7e8c60)

**Comment by rdz-oss** [comment]
Good morning,
 I have not seen this issue before. But it does look solvable. Just to confirm before I drill in, you installed the (**legacy, 64 Bit**) Lite version? 

<img width="638" alt="Screenshot 2024-03-17 at 09 35 13" src="https://github.com/rdz-oss/BattyBirdNET-Pi/assets/63958211/470d95d6-a7c7-4d9d-bd16-66d555712b08">


<img width="596" alt="Screenshot 2024-03-17 at 09 39 01" src="https://github.com/rdz-oss/BattyBirdNET-Pi/assets/63958211/4a0d42cd-5e83-4d92-ae5e-41602c98b114">

**Reply by diplopoda** [reply]
I used the downloaded file:

2024-03-12-raspios-bullseye-arm64-lite.img.xz

from here:

https://www.raspberrypi.com/software/operating-systems/

and used the installer via "custum rom".

**Reply by rdz-oss** [reply]
I will try to reproduce this. Might take some time. If you like you could try a hack in the meantime, ie
call 
sudo apt-get update
sudo apt-get upgrade
sudo apt-get install python3-venv
and then run the install script again.

**Reply by diplopoda** [reply]
Hello Richard, 

that worked like a charm!

BattyBirdNET-Pi is running and I'm testing. The first Ppip were flying around the Audiomoth. I rund the RPI with the Audiomoth via Powerbank but there's a constant humming and some strange baseline noises (>15kHz) which sometimes trigger Nynoc-determinations. The Ppip was determined as Pkuh and Pnat sometimes with pretty high confidence but I inspected the recordings and it was Ppip.

Thanks again.

**Reply by rdz-oss** [reply]
Happy to hear! Are you running UK/Eu or Bavaria classifier?
I will get around to optimizing the detection rates again sometime this year. Perhaps you might consider helping by testing and/or sending some data sometime in the future? 
It is already good in capturing the calls - ie presence or absence of a bat. Getting better step by step if people help :)
Enjoy!

**Reply by rdz-oss** [reply]
Pasted in wrong place. I have Kuhls in my detections today, perhaps you can check?
https://github.com/rdz-oss/BattyBirdNET-Pi/discussions/15#discussioncomment-8820650

**Comment by rdz-oss** [comment]
By the way you can enable/disable a timer (sundown to sunup) and a noise scrubbing functionality if you want to test under Tools - Settings - Advanced Settings

**Comment by rdz-oss** [comment]
Answered

**Comment by shawnmccracken** [comment]
I just wanted to add that I was not able to get a Raspberry Pi 4 Model B with 8GB RAM and USB AudioMoth to work using the 'easy' install method. Using the 'regular' method it all worked fine.
